testscripts/kdump_3_types_storage.ps1

```diff
@@ -0,0 +1,389 @@
+    bin\plink.exe -i ssh\${sshKey} root@$IP_A "sed -i 's?core_collector makedumpfile -F --message-level 1 -d 31?core_collector makedumpfile -l --message-level 1 -d 31?' /etc/kdump.conf"
+    bin\plink.exe -i ssh\${sshKey} root@$IP_A "echo $UUID_type $UUID >> /etc/kdump.conf"
+    $result = bin\plink.exe -i ssh\${sshKey} root@$IP_A "systemctl restart kdump ; echo `$?"
+    if ($result -ne 0)
+    {
+        Write-Host -F Red "ERROR: Restart kdump failed: $result"
+        Write-Output "ERROR: Restart kdump failed: $result"
+        return $false
+    }
+    return $true
+}
+
+
+# Function to trigger VMA
+Function TriggerVM($sshKey,${IP_A},${IP_B},$check_dir,$storage)
+{
+    Write-Host -F Green "INFO: Trigger the $ipv4 by $storage storage"
+    Start-Process ".\bin\plink.exe" "-i .\ssh\demo_id_rsa.ppk root@${IP_A} echo 1 > /proc/sys/kernel/sysrq && echo c > /proc/sysrq-trigger" -PassThru -WindowStyle Hidden
+    sleep 300
+    if ($storage -eq "UUID")
+    {
+        $crash = bin\plink.exe -i ssh\${sshKey} root@$IP_A "du -h ${check_dir}"
+    }
+    else
+    {
+        $crash = bin\plink.exe -i ssh\${sshKey} root@$IP_B "du -h ${check_dir}"
+    }
+    Write-Host -F Green "DENUG: Show the result of server: $($crash[0])"
+    Write-Output "DENUG: Show the result of server: $($crash[0])"
+    $crash[0] -match "^\d{1,3}M\b"
+    $vmcore = $($matches[0]).Substring(0,$matches[0].length-1)
+    if ([int]$vmcore -gt 30)
+    {
+        return $true
+    }
+    Write-Host -F Red "ERROR: Trigger kdump throght $storage store failed: $check_result"
+    Write-Output "ERROR: Check ${check_dir}: $vmcore"
+    return $false
+}
+
+
+# Prepare VMB
+$vmOut = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+$vmNameB = $vmName -creplace ("-A$"),"-B"
+Write-Host -F Green "INFO: RevertSnap $vmNameB..."
+Write-Output "INFO: RevertSnap $vmNameB..."
+$result = RevertSnapshotVM $vmNameB $hvServer
+if ($result[-1] -ne $true)
+{
+    Write-Host -F Red "ERROR: RevertSnap $vmNameB failed"
+    Write-Output "ERROR: RevertSnap $vmNameB failed"
+    DisconnectWithVIServer
+    return $Aborted
+}
+$vmObjB = Get-VMHost -Name $hvServer | Get-VM -Name $vmNameB
+Write-Host -F Green "INFO: Starting $vmNameB..."
+Write-Output "INFO: Starting $vmNameB..."
+Start-VM -VM $vmObjB -Confirm:$false -RunAsync:$true -ErrorAction SilentlyContinue
+$ret = WaitForVMSSHReady $vmNameB $hvServer ${sshKey} 300
+if ($ret -ne $true)
+{
+    write-host -F Red "Failed: Failed to start VM."
+    Write-Output "Failed: Failed to start VM."
+    DisconnectWithVIServer
+    return $Aborted
+}
+# Refresh status
+$vmObjB = Get-VMHost -Name $hvServer | Get-VM -Name $vmNameB
+$IPB = GetIPv4ViaPowerCLI $vmNameB $hvServer
+
+
+# Change crashkernel
+Write-Host -F Green "INFO: Change the crashkernel = 512M of $ipv4"
+$result = bin\plink.exe -i ssh\${sshKey} root@$ipv4 "sed -i 's?crashkernel=auto?crashkernel=512M?' /etc/default/grub && grub2-mkconfig -o $dir && echo `$? "
```

Comment:
If core of case isn't test crashkernel value, please use "auto", i think it is very stable than 512M.

Reply:
If keep it as "auto", when the memory of VMs<=8G,  kdump will not start. So we need to config it.

Reply:
You mean VM's memory size < 8G, crashkernel=auto won't work, right?